### ISSUE TITLE
Brass Catcher causes invalid container error

### ISSUE THREAD

**Issue by bouchacha** [CleverRaven/Cataclysm-DDA#44065]
# Describe the bug

Firing a firearm with a brass catcher attached causes this error:
```
 DEBUG    : tried to put an item (223_casing) count (1) in a container (m4a1) that cannot contain it: pocket with type (CONTAINER) not found

 FUNCTION : ret_val<bool> item::put_in(const item&, item_pocket::pocket_type)
 FILE     : src/item.cpp
 LINE     : 1103
```
It apparently also eats up the casings instead of saving them.

# Steps To Reproduce
1. Attach a brass catcher
2. Fire the weapon
3. Error!

# Expected behavior
No error should come up. Casings should be saved.

# Versions and configuration
```
- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 0.E-5799-gd2843c7 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Graphical Overmap [Graphical_Overmap],
    Stats Through Skills [StatsThroughSkills]
]
```

**Comment by mqrause** [comment]
This is the same as #42313, just the message is new (introduced in #41677). So it's also gonna be fixed by #42455.

**Comment by wapcaplet** [comment]
Tested OK in 0.E-5926-g a883429 - brass catchers work as expected for both a .45 USP and an M2010 ESR. Closing issue.
